src/GraphQL.Tests/Subscription/ShouldlyExtensions.cs

```diff
@@ -57,4 +58,16 @@ public static ExecutionResult ShouldBeSuccessful(this ExecutionResult result)
         result.Executed.ShouldBeTrue();
         return result;
     }
+
+    public static void ShouldBeWhen(this object? obj, bool condition, object onTrue, object onFalse) =>
+        obj.ShouldBe(condition ? onTrue : onFalse);
+
+    public static void ShouldThrowWhen<TException>(bool condition, Action action)
+        where TException : Exception
+    {
+        if (condition)
+            Should.Throw<TException>(action);
+        else
+            Should.NotThrow(action);
```

Comment:
I would not call `Should.NotThrow`; I would call the action directly.  If it throws, the stack trace and error will show up in the test explorer.  But if you have a reason for doing so, that's fine; whatever you think.